mypy/fscache.py

```diff
@@ -230,6 +230,7 @@ def exists_case(self, path: str, prefix: str) -> bool:
         if path in self.exists_case_cache:
             return self.exists_case_cache[path]
         head, tail = os.path.split(path)
+        prefix = prefix.rstrip(os.sep)
         if not head.startswith(prefix) or not tail:
```

Comment:
Hm, sort of pre-existing, but I guess stripping is a little problematic for startswith, e.g. `path=/foobar/baz, prefix=/foo`. I guess it just means there ends up being another listdir

Reply:
Ah yes, didn't think about that before. That's indeed pre-existing, but changed the condition a bit to catch that as well.